### PR TITLE
#minor Enable hostnetworking globablly for all launched pods

### PR DESCRIPTION
**Pull request by kumare3**
Signed-off-by: Ketan Umare <ketan.umare@gmail.com>

# TL;DR
This PR adds a new flag that can allow admins to enable host networking for the entire cluster. 
This should be superseded by - https://github.com/flyteorg/flyte/issues/2069

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
NA

### CODE REVIEW

**Review by EngHabu — APPROVED** [review]
Sad.. I think we should work on deprecating all of these into PodTemplate object as soon as possible...

**Review by hamersaw — DISMISSED** [review]

File: go/tasks/config_load_test.go [hunk 1]
```diff
@@ -108,4 +110,4 @@ func TestLoadConfig(t *testing.T) {
 	t.Run("sagemaker-config-test", func(t *testing.T) {
 		assert.NotNil(t, sagemakerConfig.GetSagemakerConfig())
 	})
-}
+}
```
[the remark this answers is not in the document]
Comment (kumare3) [reply go/tasks/config_load_test.go:113]: Will do

File: go/tasks/pluginmachinery/flytek8s/config/config.go [hunk 2]
```diff
@@ -139,6 +139,11 @@ type K8sPluginConfig struct {
 	// DefaultSecurityContext provides a default container security context that should be applied for the primary container launched and created by FlytePropeller. This may not be applicable to all plugins. For
 	//	// downstream plugins - i.e. TensorflowOperators may not support setting this, but Spark does.
 	DefaultSecurityContext *v1.SecurityContext `json:"default-security-context" pflag:"-,Optionally specify a default security context that should be applied to every container launched/created by FlytePropeller. This will not be applied to plugins that do not support it or to user supplied containers in pod tasks."`
+
+	// EnableHostNetworkingPod is a binary switch to enable `hostNetwork: true` for all pods launched by Flyte.
+	// Refer to - https://kubernetes.io/docs/concepts/policy/pod-security-policy/#host-namespaces.
+	// As a follow up, the default pod configurations will now be adjusted using podTemplates per namespace
+	EnableHostNetworkingPod *bool `json:"enable-host-networking-pod" pflag:"-,If true, will schedule all pods with hostNetwork: true."`
```
Comment (hamersaw) [inline go/tasks/pluginmachinery/flytek8s/config/config.go:146]: Also, just noted on flytepropeller testing. We can remove the "-" and generate a pflag for this right?

Comment (kumare3) [reply go/tasks/pluginmachinery/flytek8s/config/config.go:146]: we can remove, i dont want to, as i want to disable this

**Review by pmahindrakar-oss — APPROVED** [review]

**Pull request merged**